hpx/parallel/algorithms/inner_product.hpp

```diff
@@ -290,22 +81,9 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
     inner_product(ExPolicy&& policy, InIter1 first1, InIter1 last1,
         InIter2 first2, T init)
     {
```

Comment:
Hmmm. Ok, I see the point. No use in duplicating this code. Would have been nice to have errors as early as possible in the call chain.